plugins/module_utils/ldap.py

```diff
@@ -139,5 +139,5 @@ def _connect_to_ldap(self):
 
     def _xorder_dn(self):
         # match X_ORDERed DNs
-        regex = r"\w+=\{\d+\}.+"
+        regex = r".+\{\d+\}.+"
```

Comment:
Now it also matches `some random text {1} more text`. That looks way too general to me.

Reply:
Yes, that's the purpose of this function, **but** it should avoid false positives. Your regex makes false positives a lot more likely.

Reply:
I'm not quite sure I follow, sorry. Do you have an example where the function would return `True` for a valid DN which does not contain an order number?

I realise it will match on random strings which contain that sequence, but unless we are checking for valid DN syntax, that will always be the case.

Reply:
Ok, I see.

How about modifying the check slightly by first calling `ldap.dn.explode_dn(dn)` and only checking whether the first component contains `{\d+\}` (with `re.search`)?

Reply:
Sounds fine to me.